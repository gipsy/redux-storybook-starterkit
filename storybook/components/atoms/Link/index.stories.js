import React from 'react'
import { storiesOf } from '@storybook/react'
import Link from '~/src/components/atoms/Link'

storiesOf('Components/Atoms', module)
  .add('Link', () => (
    <Link href="https://github.com/diegohaz/arc">ARc repository</Link>
  ))
