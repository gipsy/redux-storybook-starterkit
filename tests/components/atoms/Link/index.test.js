import React from 'react'
import { shallow } from 'enzyme'
import Link from '~/src/components/atoms/Link'

const wrap = (props = {}) => shallow(<Link {...props} />).dive()

it('renders children when passed in', () => {
  const wrapper = wrap({ children: 'test' })
  expect(wrapper.contains('test')).toBe(true)
})

it('renders props when passed in', () => {
  const wrapper = wrap({ id: 'foo' })
  expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
})

it('renders anchor by default', () => {
  const wrapper = wrap()
  expect(wrapper.find('a')).toHaveLength(1)
})

it('renders Link when prop to is passed in', () => {
  const wrapper = wrap({ to: 'test' }).dive()
  expect(wrapper.find('NavLink')).toHaveLength(1)
})

test('<Link /> renders with palette: primary as prop', () => {
  const wrapper = wrap({ palette: 'primary' })
  expect(wrapper).toMatchSnapshot()
})
