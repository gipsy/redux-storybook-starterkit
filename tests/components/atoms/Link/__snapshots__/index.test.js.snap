// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders with palette: primary as prop 1`] = `
<a
  className="sc-bwzfXH cPzigO"
/>
`;
